### ISSUE TITLE
Wrong regex (proxy variable in compose.pp)

### ISSUE THREAD

**Issue by Jeoffreybauvin** [puppetlabs/puppetlabs-docker#203]
## What you expected to happen?

I'm using :

`docker::proxy: 'http://prod.all.httpproxy:3128'`

## What happened?

`Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, validate_re(): "http://myproxy:3128" does not match "^((http[s]?)?:\\/\\/)?([^:^@]+:[^:^@]+@|)([\\da-z\\.-]+)\\.([\\da-z\\.]{2,6})(:[\\d])?([\\/\\w \\.-]*)*\\/?$" at /etc/puppetlabs/code/environments/staging/modules/docker/manifests/compose.pp:31:7 on node mynode`

## How to reproduce it?

If  I use "http://myproxy", it's working. The regexp is wrong.

## Versions:
Puppet-docker 1.0.4, but same behavior with last version

## Logs:
```
Paste any relevant logs from a puppet run
```

**Comment by davejrt** [comment]
The error you've posted, and the configuration refer to two different proxy settings. The first is the proxy for the docker daemon, and the log is from the compose class which only gets used when downloading compose from github. 

In any case, with the following config I cannot replicate your error using 1.1.0 of the module. 

```class  {'docker':} ->

class {'docker::compose':
  ensure => present,
  version => '1.18.0',
  proxy => 'https://test.com:3128',

}```

I get an error timing out which I'd expect as this proxy is fake, but I don't get the regex error. Can you please edit your initial issue with all the info requested in the template like manifests etc?

**Comment by scotty-c** [comment]
@Jeoffreybauvin Is this issue still valid. I Believe this was fixed in the v1.1.0 release

**Comment by scotty-c** [comment]
closing due to inactivity

**Comment by olivierHa** [comment]
Hello @scotty-c , Could you re-opened it ?

For me this isn't fixed, even in 3.0.1

@davejrt please could you try with a proxy like https://foo.bar.test.com:3128 and not "only" test.com ?

Regards

Olivier

**Comment by davejrt** [comment]
@olivierHa see below, still cannot replicate any issue related, I only get a timeout.

```root@node-01:~# puppet module list --modulepath /tmp/modules
/tmp/modules
├── camptocamp-kmod (v2.2.0)
├── davejrt-run_container (v0.1.0)
├── herculesteam-augeasproviders_core (v2.2.0)
├── herculesteam-augeasproviders_sysctl (v2.2.1)
├── puppetlabs-apt (v6.2.1)
├── puppetlabs-docker (v3.1.0)
├── puppetlabs-powershell (v2.2.0)
├── puppetlabs-reboot (v2.0.0)
├── puppetlabs-stdlib (v5.1.0)
└── puppetlabs-translate (v1.1.0)
root@node-01:~#```

```  class {'docker::compose':
    ensure => present,
    version => '1.23.1',
    proxy => 'https://foo.bar.test.com:3128',
  }```

```==> node-01: Debug: Executing: 'curl -s -S -L --proxy https://foo.bar.test.com:3128 https://github.com/docker/compose/releases/download/1.23.1/docker-compose-Linux-x86_64 -o /usr/local/bin/docker-compose-1.23.1'
==> node-01: Notice: /Stage[main]/Docker::Compose/Exec[Install Docker Compose 1.23.1]/returns: curl: (7) Failed to connect to foo.bar.test.com port 3128: Connection timed out
==> node-01: Error: 'curl -s -S -L --proxy https://foo.bar.test.com:3128 https://github.com/docker/compose/releases/download/1.23.1/docker-compose-Linux-x86_64 -o /usr/local/bin/docker-compose-1.23.1' returned 7 instead of one of [0]
==> node-01: Error: /Stage[main]/Docker::Compose/Exec[Install Docker Compose 1.23.1]/returns: change from 'notrun' to ['0'] failed: 'curl -s -S -L --proxy https://foo.bar.test.com:3128 https://github.com/docker/compose/releases/download/1.23.1/docker-compose-Linux-x86_64 -o /usr/local/bin/docker-compose-1.23.1' returned 7 instead of one of [0]
==> node-01: Notice: /Stage[main]/Docker::Compose/File[/usr/local/bin/docker-compose-1.23.1]: Dependency Exec[Install Docker Compose 1.23.1] has failures: true```
